particle-iot/device-os
# Store test coverage to an HTML report

### Problem

We're not using Coveralls anymore to store and track test coverage. Going forward, we want an HTML coverage report to be available in CircleCI.

### Solution

Generate an HTML coverage report in CI and stop reporting coverage to Coveralls.

This report will be stored in CircleCI with the step.
```
store_artifacts:
  path: coverage
```

### Steps to Test

Run `ci/unit_tests.sh` and expect to see a new directory `build/coverage` with files like:
```
coverage_details.css
coverage_details.c_string.h.3feed289d9c4dcffe2c42a6b9d8544f2.html
coverage_details.endian_util.h.ad0383ed740dca731e440d8dea2d3d2b.html
coverage_details.enumflags.h.66c518edb73e91dd9a2134edcdc40f02.html
coverage_details.events.cpp.a11b4b4d1b3158270da6505d53a48820.html
coverage_details.firmware_update.cpp.9f9f707748ebf23ac05a7483d9515676.html
coverage_details.firmware_update.h.148bbedd196d7e872890249a513ff66b.html
coverage_details.functions.h.b966f83160e58f49f8883abc4e51d79a.html
coverage_details.gthr-default.h.545b722277a99f42c738eac6f4eca8d7.html
coverage_details.hex_to_bytes.h.cb0f7ffe2fe0f055cf6f38f0f618bcf3.html
coverage_details.html
```

Open `coverage_details.html` to view the coverage report.
